conans/client/cmd/download.py

```diff
@@ -11,7 +11,15 @@ def download(ref, package_ids, remote, recipe, remote_manager,
 
     hook_manager.execute("pre_download", reference=ref, remote=remote)
 
-    ref = remote_manager.get_recipe(ref, remote)
+    try:
+        ref = remote_manager.get_recipe(ref, remote)
+    except NotFoundException as exc:
+        if exc.args and "404" in exc.args[0]:
```

Comment:
You can skip the check of 404, you already got that exception. It is ok to re-raise it with a new message.